sys/net/application_layer/ng_zep/ng_zep.c

```diff
@@ -0,0 +1,1072 @@
+                return -ENOTSUP;
+            }
+
+            dev->chan = *_get_uint16_ptr(value);
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_ADDRESS:
+            if (value_len < sizeof(be_uint16_t)) {
+                return -EOVERFLOW;
+            }
+            else {
+                be_uint16_t *val = value;
+
+                dev->addr = byteorder_btols(*val);
+                return sizeof(be_uint16_t);
+            }
+
+        case NETCONF_OPT_ADDRESS_LONG:
+            if (value_len < sizeof(be_uint64_t)) {
+                return -EOVERFLOW;
+            }
+            else {
+                be_uint64_t *val = value;
+
+                dev->eui64 = byteorder_btolll(*val);
+                return sizeof(be_uint64_t);
+            }
+
+        case NETCONF_OPT_ADDR_LEN:
+            if (value_len < sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+
+            switch (*_get_uint16_ptr(value)) {
+                case 2:
+                    dev->flags &= ~NG_ZEP_FLAGS_DST_ADDR_LONG;
+                    break;
+
+                case 8:
+                    dev->flags |= NG_ZEP_FLAGS_DST_ADDR_LONG;
+                    break;
+
+                default:
+                    return -ENOTSUP;
+            }
+
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_SRC_LEN:
+            if (value_len < sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+
+            switch (*_get_uint16_ptr(value)) {
+                case 2:
+                    dev->flags &= ~NG_ZEP_FLAGS_SRC_ADDR_LONG;
+                    break;
+
+                case 8:
+                    dev->flags |= NG_ZEP_FLAGS_SRC_ADDR_LONG;
+                    break;
+
+                default:
+                    return -ENOTSUP;
+            }
+
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_NID:
+            if (value_len < sizeof(be_uint16_t)) {
+                return -EOVERFLOW;
+            }
+            else {
+                be_uint16_t *val = value;
+
+                dev->pan = byteorder_btols(*val);
+                return sizeof(be_uint16_t);
+            }
+
+        case NETCONF_OPT_AUTOACK:
+            if (value_len < sizeof(ng_netconf_enable_t)) {
+                return -EOVERFLOW;
+            }
+
+            _set_flag_ptr(value, dev->flags, NG_ZEP_FLAGS_AUTOACK);
+            return sizeof(uint16_t);
+
+        default:
+            return -ENOTSUP;
+    }
+}
+
+static void _isr_event(ng_netdev_t *dev, uint32_t event_type)
+{
+    switch (event_type) {
+        case _EVENT_RX_STARTED:
+            DEBUG("zep: ISR event: RX started\n");
+            _rx_started_event((ng_zep_t *)dev);
+            break;
+
+        default:
+            DEBUG("zep: event %" PRIu32 " not handled\n", event_type);
+            break;
+    }
+}
+
+void *_event_loop(void *args)
+{
+    msg_t msg, ack, msg_q[NG_ZEP_MSG_QUEUE_SIZE];
```

Comment:
Yes, since all other event loops in the network stack are allocated this way, I thought we came to the conclusion to do it this way.